Makefile

```diff
@@ -1,38 +1,85 @@
-CC=gcc
-CXX=g++
-NVCC=nvcc
+# Load site-specific setting -- controlled using environment variable `site`:
```

Comment:
I am fine with these new make options. It's clean,and it we're helping users with various setups, great. Don't forget to document these make options in the docs and the makefile. Note in the FINUFFT makefile I did it the other way round, ie set defaults via :=  and then -includes the make.inc.  All good.

Reply:
Where would you like me to document this? The README (in lieu of an actual docs -- like the one for NUFFT?)

Reply:
I think the confusion is because cufinufft's documentation is not as comprehensive as FINUFFT.  I took a pass at docs for the Python code (sphinx), but we just have the README for the top level code here. I can imagine one day the documentation being comprehensive for the top level (can just include the python docs then) but that is out of scope for this PR.

It might be reasonable to add an `Advanced Makefile Usage` or something similar as a level 2/3 heading in the README, depending how much detail it requires to be clear for your users.

Reply:
Done -- I've put a wee doc into the README

Reply:
Looks very serviceable, thank you for doing that.

@MelodyShih , note this will change location of the make.inc file I think you used.. so you're not surprised.

Reply:
Got it. Thanks for letting me know.